src/renderer/src/components/playlist/playlist-song-list/PlaylistSongList.tsx

```diff
@@ -1,76 +1,74 @@
-import { ArrowLeftIcon, DeleteIcon, PencilIcon, PencilOffIcon, Trash2Icon } from "lucide-solid";
-import { Component, createSignal, Match, onCleanup, onMount, Show, Switch } from "solid-js";
-import { PlaylistSongsQueryPayload, ResourceID, Song } from "src/@types";
+import { ArrowLeftIcon, DeleteIcon, PencilIcon } from "lucide-solid";
+import { Component, createSignal, onCleanup, onMount } from "solid-js";
+import { Playlist, PlaylistSongsQueryPayload, ResourceID, Song } from "src/@types";
 
 type PlaylistSongListProps = {
-  playlistName: string;
+  playlist: Playlist;
 };
 
 const PlaylistSongList: Component<PlaylistSongListProps> = (props) => {
   const group = namespace.create(true);
 
   const [payload] = createSignal<PlaylistSongsQueryPayload>({
-    playlistName: props.playlistName,
+    playlistName: props.playlist.name,
   });
 
-  const [editMode, setEditMode] = createSignal(false);
   const [, setIsQueueExist] = createSignal(false);
 
   const reset = new Impulse();
 
   onMount(async () => {
     window.api.listen("playlist::resetSongList", reset.pulse.bind(reset));
     setIsQueueExist(await window.api.request("queue::exists"));
-  });
 
-  onCleanup(() => window.api.removeListener("playlist::resetSongList", reset.pulse.bind(reset)));
+    onCleanup(() => {
+      window.api.removeListener("playlist::resetSongList", reset.pulse.bind(reset));
+    });
+  });
 
   const createQueue = async (songResource: ResourceID) => {
     await window.api.request("queue::create", {
       startSong: songResource,
       order: { direction: "asc", option: "none" },
       tags: [],
-      view: { playlist: props.playlistName },
+      view: { playlist: props.playlist.name },
     });
     setIsQueueExist(true);
   };
 
   return (
     <div class="flex h-full flex-col">
       <div class="sticky top-0 z-10 mx-5 mb-4 mt-1 flex flex-row items-center">
-        <div class="flex w-full flex-row items-center gap-5 font-medium">
+        <div class="flex w-full flex-row items-center gap-2.5 font-medium">
           <Button
-            variant={"ghost"}
-            size={"square"}
-            onClick={() => setPlaylistActiveScene(PLAYLIST_SCENE_LIST)}
+            variant="ghost"
+            size="square"
+            onClick={() => setPlaylistActivePage({ name: "list" })}
           >
             <ArrowLeftIcon class="text-subtext" size={20} />
           </Button>
-          <h3 class="text-xl">{props.playlistName}</h3>
+          <h3 class="text-xl uppercase">{props.playlist.name}</h3>
         </div>
         <Button
-          variant={editMode() ? "secondary" : "outlined"}
+          variant="outlined"
           size={"square"}
           class="rounded-lg"
           onClick={() => {
-            setEditMode(!editMode());
+            setPlaylistActivePage({
+              name: "edit",
+              playlist: props.playlist,
+              from: { name: "songs", playlist: props.playlist },
+            });
```

Comment:
If you click the edit button in this page and change the name of the playlist it puts you back in a page with the old playlist name.
If you edit the image of the playlist and then re-click the edit button you will see the old image.

Reply:
Fixed this issue.